diff --git a/node_modules/vuepress-theme-vdoing/components/Buttons.vue b/node_modules/vuepress-theme-vdoing/components/Buttons.vue
index da2fbb4..184ecc9 100644
--- a/node_modules/vuepress-theme-vdoing/components/Buttons.vue
+++ b/node_modules/vuepress-theme-vdoing/components/Buttons.vue
@@ -14,40 +14,11 @@
       v-show="showCommentBut"
       @click="scrollToComment"
     />
-    <div
-      title="主题模式"
-      class="button blur theme-mode-but iconfont icon-zhuti"
-      @mouseenter="showModeBox = true"
-      @mouseleave="showModeBox = false"
-      @click="showModeBox = true"
-    >
-      <transition name="mode">
-        <ul
-          class="select-box"
-          ref="modeBox"
-          v-show="showModeBox"
-          @click.stop
-          @touchstart.stop
-        >
-          <li
-            v-for="item in modeList"
-            :key="item.KEY"
-            class="iconfont"
-            :class="[item.icon, { active: item.KEY === currentMode }]"
-            @click="toggleMode(item.KEY)"
-          >
-            {{ item.name }}
-          </li>
-        </ul>
-      </transition>
-    </div>
   </div>
 </template>
 
 <script>
 import debounce from 'lodash.debounce'
-import storage from 'good-storage' // 本地存储
-const MOBILE_DESKTOP_BREAKPOINT = 719 // refer to config.styl
 
 export default {
   data () {
@@ -56,30 +27,6 @@ export default {
       scrollTop: null,
       showCommentBut: false,
       commentTop: null,
-      currentMode: null,
-      showModeBox: false,
-      modeList: [
-        {
-          name: '跟随系统',
-          icon: 'icon-zidong',
-          KEY: 'auto'
-        },
-        {
-          name: '浅色模式',
-          icon: 'icon-rijianmoshi',
-          KEY: 'light'
-        },
-        {
-          name: '深色模式',
-          icon: 'icon-yejianmoshi',
-          KEY: 'dark'
-        },
-        {
-          name: '阅读模式',
-          icon: 'icon-yuedu',
-          KEY: 'read'
-        }
-      ],
       _scrollTimer: null,
       _textareaEl: null,
       _recordScrollTop: null,
@@ -89,8 +36,6 @@ export default {
     }
   },
   mounted () {
-    this.currentMode = storage.get('mode') || 'auto'
-
     this.scrollTop = this.getScrollTop()
     window.addEventListener('scroll', debounce(() => {
       this.scrollTop = this.getScrollTop()
@@ -100,20 +45,6 @@ export default {
       this.getCommentTop()
     })
 
-    // 小屏时选择主题模式后关闭选择框
-    if (document.documentElement.clientWidth < MOBILE_DESKTOP_BREAKPOINT) {
-      const modeBox = this.$refs.modeBox
-      modeBox.onclick = () => {
-        this.showModeBox = false
-      }
-      window.addEventListener('scroll', debounce(() => {
-        if (this.showModeBox) {
-          this.showModeBox = false
-        }
-      }, 100))
-    }
-
-
     // 移动端对类似:hover效果的处理
     const buttons = document.querySelectorAll('.buttons .button')
     for (let i = 0; i < buttons.length; i++) {
@@ -135,10 +66,6 @@ export default {
     }
   },
   methods: {
-    toggleMode (key) {
-      this.currentMode = key
-      this.$emit('toggle-theme-mode', key)
-    },
     getScrollTop () {
       return window.pageYOffset
         || document.documentElement.scrollTop
diff --git a/node_modules/vuepress-theme-vdoing/components/Footer.vue b/node_modules/vuepress-theme-vdoing/components/Footer.vue
index 1b5e5c0..54e28b5 100644
--- a/node_modules/vuepress-theme-vdoing/components/Footer.vue
+++ b/node_modules/vuepress-theme-vdoing/components/Footer.vue
@@ -42,6 +42,8 @@ export default {
 <style lang='stylus'>
 // $mobileSidebarWidth = $sidebarWidth * 0.82
 .footer
+  position absolute
+  width: 100%
   padding 5rem 1.5rem 2.5rem
   text-align center
   color #666
@@ -61,12 +63,12 @@ export default {
       color $accentColor
 @media (min-width ($MQMobile + 1px))
   .sidebar-open .footer
-    width auto
+    width 100%
     padding-left: ($sidebarWidth + 1.5rem)
 @media (min-width 1520px)
   .have-rightmenu .footer
     padding-right: ($rightMenuWidth + 1.5rem)
 .no-sidebar .footer
-  width auto
+  width 100%
   padding-left 1.5rem
 </style>
diff --git a/node_modules/vuepress-theme-vdoing/components/Home.vue b/node_modules/vuepress-theme-vdoing/components/Home.vue
index d316ee0..1499b13 100644
--- a/node_modules/vuepress-theme-vdoing/components/Home.vue
+++ b/node_modules/vuepress-theme-vdoing/components/Home.vue
@@ -6,6 +6,7 @@
       :class="{ 'hide-banner': !showBanner }"
       :style="bannerBgStyle"
     >
+      <video v-show="videoBackgroundSrc != null" :src="videoBackgroundSrc" class="video-background" autoPlay preload="auto" loop="loop" muted="muted" controlsList="nodownload" oncontextmenu="return(false);"></video>
       <div
         class="banner-conent"
         :style="
@@ -14,6 +15,7 @@
       >
         <header class="hero">
           <img
+            class="hero-image"
             v-if="homeData.heroImage"
             :src="$withBase(homeData.heroImage)"
             :alt="homeData.heroAlt"
@@ -24,13 +26,16 @@
           <p v-if="homeData.tagline" class="description">
             {{ homeData.tagline }}
           </p>
-          <p class="action" v-if="homeData.actionText && homeData.actionLink">
-            <NavLink class="action-button" :item="actionLink" />
+          <div v-if="homeData.slogan" v-html="homeData.slogan"></div>
+          <p class="action" v-if="hasActions">
+            <NavLink v-for="(action, index) in homeData.actions"  :key="index"
+              :class="action.actionClass || 'action-button'"
+              :item="{text: action.actionText, link: action.actionLink}" />
           </p>
         </header>
 
         <!-- PC端features块 s -->
-        <div class="features" v-if="hasFeatures && !isMQMobile">
+        <div class="features" v-if="hasFeatures" v-show="!isMQMobile">
           <div
             class="feature"
             v-for="(feature, index) in homeData.features"
@@ -159,7 +164,6 @@
     </MainLayout>
   </div>
 </template>
-
 <script>
 import NavLink from "@theme/components/NavLink";
 import BScroll from "@better-scroll/core"
@@ -179,7 +183,7 @@ BScroll.use(Slide)
 export default {
   data() {
     return {
-      isMQMobile: false,
+      isMQMobile: true,
 
       slide: null,
       currentPageIndex: 0,
@@ -191,7 +195,26 @@ export default {
       currentPage: 1// 当前页
     }
   },
+  props: {
+    themeMode: {
+      type: String,
+      required: true,
+      default: null
+    }
+  },
   computed: {
+    videoBackgroundSrc() {
+      if (this.isMQMobile){
+        return null;
+      }
+      if (this.themeMode === 'dark'){
+        return "/video/bg-dark.mov"
+      }else if (this.themeMode === 'light'){
+        return "/video/bg-light.mov"
+      }else{
+        return null;
+      }
+    },
     homeData() {
       return {
         ...this.$page.frontmatter
@@ -200,6 +223,9 @@ export default {
     hasFeatures() {
       return !!(this.homeData.features && this.homeData.features.length)
     },
+    hasActions() {
+      return !!(this.homeData.actions && this.homeData.actions.length)
+    },
     homeSidebarB() {
       const { htmlModules } = this.$themeConfig
       return htmlModules ? htmlModules.homeSidebarB : ''
@@ -230,12 +256,6 @@ export default {
         return `background: url(${this.$withBase(bannerBg)}) center center / cover no-repeat`
       }
 
-    },
-    actionLink() {
-      return {
-        link: this.homeData.actionLink,
-        text: this.homeData.actionText
-      };
     }
   },
   components: { NavLink, MainLayout, PostList, UpdateArticle, BloggerBar, CategoriesBar, TagsBar, Pagination },
@@ -246,6 +266,7 @@ export default {
     this.isMQMobile = window.innerWidth < MOBILE_DESKTOP_BREAKPOINT ? true : false; // vupress在打包时不能在beforeCreate(),created()访问浏览器api（如window）
   },
   mounted() {
+
     if (this.$route.query.p) {
       this.currentPage = Number(this.$route.query.p)
     }
diff --git a/node_modules/vuepress-theme-vdoing/components/NavLink.vue b/node_modules/vuepress-theme-vdoing/components/NavLink.vue
index 93feac7..429934b 100644
--- a/node_modules/vuepress-theme-vdoing/components/NavLink.vue
+++ b/node_modules/vuepress-theme-vdoing/components/NavLink.vue
@@ -11,8 +11,8 @@
     :href="link"
     @focusout="focusoutAction"
     class="nav-link external"
-    :target="isMailto(link) || isTel(link) ? null : '_blank'"
-    :rel="isMailto(link) || isTel(link) ? null : 'noopener noreferrer'"
+    :target="target"
+    :rel="rel"
   >
     {{ item.text }}
     <OutboundLink />
@@ -39,6 +39,29 @@ export default {
         return Object.keys(this.$site.locales).some(rootLink => rootLink === this.link)
       }
       return this.link === '/'
+    },
+
+    target () {
+      if (this.isNonHttpURI) {
+        return null
+      }
+      if (this.item.target) {
+        return this.item.target
+      }
+      return isExternal(this.link) ? '_blank' : ''
+    },
+    
+    rel () {
+      if (this.isNonHttpURI) {
+        return null
+      }
+      if (this.item.rel === false) {
+        return null
+      }
+      if (this.item.rel) {
+        return this.item.rel
+      }
+      return this.isBlankTarget ? 'noopener noreferrer' : null
     }
   },
 
diff --git a/node_modules/vuepress-theme-vdoing/components/NavLinks.vue b/node_modules/vuepress-theme-vdoing/components/NavLinks.vue
index 7be2ca3..1eaa2ec 100644
--- a/node_modules/vuepress-theme-vdoing/components/NavLinks.vue
+++ b/node_modules/vuepress-theme-vdoing/components/NavLinks.vue
@@ -1,7 +1,7 @@
 <template>
   <nav
     class="nav-links"
-    v-if="userLinks.length || repoLink"
+    v-if="userLinks.length || repoLink || canToggleTheme"
   >
     <!-- user links -->
     <div
@@ -30,6 +30,16 @@
       {{ repoLabel }}
       <OutboundLink />
     </a>
+
+    <button
+      v-if="canToggleTheme"
+      type="button"
+      class="nav-theme-switch iconfont"
+      :class="themeModeIcon"
+      :title="themeToggleLabel"
+      :aria-label="themeToggleLabel"
+      @click="$emit('toggle-theme-mode', nextThemeMode)"
+    />
   </nav>
 </template>
 
@@ -41,6 +51,13 @@ import NavLink from '@theme/components/NavLink.vue'
 export default {
   components: { NavLink, DropdownLink },
 
+  props: {
+    themeMode: {
+      type: String,
+      default: null
+    }
+  },
+
   computed: {
     userNav () {
       return this.$themeLocaleConfig.nav || this.$site.themeConfig.nav || []
@@ -112,6 +129,22 @@ export default {
       }
 
       return 'Source'
+    },
+
+    canToggleTheme () {
+      return this.themeMode === 'light' || this.themeMode === 'dark'
+    },
+
+    nextThemeMode () {
+      return this.themeMode === 'dark' ? 'light' : 'dark'
+    },
+
+    themeModeIcon () {
+      return this.themeMode === 'dark' ? 'icon-yejianmoshi' : 'icon-rijianmoshi'
+    },
+
+    themeToggleLabel () {
+      return this.themeMode === 'dark' ? '切换到浅色模式' : '切换到深色模式'
     }
   }
 }
diff --git a/node_modules/vuepress-theme-vdoing/components/Navbar.vue b/node_modules/vuepress-theme-vdoing/components/Navbar.vue
index 2fb8bda..075f311 100644
--- a/node_modules/vuepress-theme-vdoing/components/Navbar.vue
+++ b/node_modules/vuepress-theme-vdoing/components/Navbar.vue
@@ -33,7 +33,11 @@
       <SearchBox
         v-else-if="$site.themeConfig.search !== false && $page.frontmatter.search !== false"
       />
-      <NavLinks class="can-hide" />
+      <NavLinks
+        class="can-hide"
+        :themeMode="themeMode"
+        @toggle-theme-mode="$emit('toggle-theme-mode', $event)"
+      />
     </div>
   </header>
 </template>
@@ -47,6 +51,13 @@ import NavLinks from '@theme/components/NavLinks.vue'
 export default {
   components: { SidebarButton, NavLinks, SearchBox, AlgoliaSearchBox },
 
+  props: {
+    themeMode: {
+      type: String,
+      required: true
+    }
+  },
+
   data () {
     return {
       linksWrapMaxWidth: null
diff --git a/node_modules/vuepress-theme-vdoing/components/Sidebar.vue b/node_modules/vuepress-theme-vdoing/components/Sidebar.vue
index 5e5c9c0..f97615d 100644
--- a/node_modules/vuepress-theme-vdoing/components/Sidebar.vue
+++ b/node_modules/vuepress-theme-vdoing/components/Sidebar.vue
@@ -1,5 +1,6 @@
 <template>
   <aside class="sidebar">
+    <slot name="top" />
     <div class="blogger" v-if="blogger">
       <img :src="blogger.avatar" />
       <div class="blogger-info">
@@ -22,7 +23,7 @@
     <!-- 移动端Nav -->
     <NavLinks />
 
-    <slot name="top" />
+
 
     <SidebarLinks :depth="0" :items="items" />
     <slot name="bottom" />
@@ -97,13 +98,22 @@ export default {
     margin-bottom -0.5rem
     font-size 0.85rem
     &.sidebar-slot-top
-      padding 1.5rem 1.5rem 0
+      padding 1.5rem;
     &.sidebar-slot-bottom
       padding 0 1.5rem 1.5rem
 @media (max-width $MQMobile)
   .sidebar
+    display: block !important;
+    overflow-y: auto !important;
+    ul[class="sidebar-links"]
+      overflow-y: hidden !important;
+      height: auto !important
+    nav[class="nav-links"]
+      overflow-y: hidden !important;
+      height: auto !important;
     .blogger
       display flex
+    .sidebar-links
     .nav-links
       display block
       .dropdown-wrapper .nav-dropdown .dropdown-item a.router-link-active::after
diff --git a/node_modules/vuepress-theme-vdoing/layouts/Layout.vue b/node_modules/vuepress-theme-vdoing/layouts/Layout.vue
index 3821782..0e710e7 100644
--- a/node_modules/vuepress-theme-vdoing/layouts/Layout.vue
+++ b/node_modules/vuepress-theme-vdoing/layouts/Layout.vue
@@ -5,7 +5,12 @@
     @touchstart="onTouchStart"
     @touchend="onTouchEnd"
   >
-    <Navbar v-if="shouldShowNavbar" @toggle-sidebar="toggleSidebar" />
+    <Navbar
+      v-if="shouldShowNavbar"
+      :themeMode="themeMode"
+      @toggle-sidebar="toggleSidebar"
+      @toggle-theme-mode="toggleThemeMode"
+    />
 
     <div class="sidebar-mask" @click="toggleSidebar(false)"></div>
 
@@ -21,12 +26,14 @@
     >
       <template #top v-if="sidebarSlotTop">
         <div
+          id="sidebar-slot-top"
           class="sidebar-slot sidebar-slot-top"
           v-html="sidebarSlotTop"
         ></div>
       </template>
       <template #bottom v-if="sidebarSlotBottom">
         <div
+          id="sidebar-slot-bottom"
           class="sidebar-slot sidebar-slot-bottom"
           v-html="sidebarSlotBottom"
         ></div>
@@ -36,7 +43,7 @@
     </Sidebar>
 
     <!-- 首页 -->
-    <Home v-if="$page.frontmatter.home" />
+    <Home v-if="$page.frontmatter.home" :themeMode="themeMode" />
 
     <!-- 分类页 -->
     <CategoriesPage v-else-if="$page.frontmatter.categoriesPage" />
@@ -67,7 +74,7 @@
 
     <Footer />
 
-    <Buttons ref="buttons" @toggle-theme-mode="toggleThemeMode" />
+    <Buttons ref="buttons" />
 
     <BodyBgImg v-if="$themeConfig.bodyBgImg" />
 
@@ -217,7 +224,10 @@ export default {
   beforeMount() {
     this.isSidebarOpenOfclientWidth()
     const mode = storage.get('mode') // 不放在created是因为vuepress不能在created访问浏览器api，如window
-    if (!mode || mode === 'auto') { // 当未切换过模式，或模式处于'跟随系统'时
+    if (mode === 'read') {
+      this._autoMode()
+      storage.set('mode', 'auto')
+    } else if (!mode || mode === 'auto') { // 当未切换过模式，或模式处于'跟随系统'时
       this._autoMode()
     } else {
       this.themeMode = mode
@@ -304,6 +314,11 @@ export default {
       }
     },
     toggleThemeMode(key) {
+      if (key === 'read') {
+        this._autoMode()
+        storage.set('mode', 'auto')
+        return
+      }
       if (key === 'auto') {
         this._autoMode()
       } else {
